.github/workflows/codeql.yml

```diff
@@ -0,0 +1,76 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
```

Comment:
I am not sure if we want to run the static analysis on every PR. Can you provide a link to a test run of the workflow? What if there is a false-positive? How can I define an exception?

Reply:
PR can be merged even with a check failing, but It's not recommended and should be reported eg. 
 https://github.com/github/codeql/issues/10766

here is an example run:
https://github.com/laDok8/sunstone/actions/runs/4165030065
vulnerabilities results are shown in the security tab of the repository, but they are private so I cannot link them